### PR TITLE
remove history line in select when user insert choice

### PR DESCRIPTION
**Pull request by albertored**
Without this change the history get polluted of user select choiches

### CODE REVIEW

**Comment by tleonhardt** [comment]
Weird, it doesn't seem like the code you added should have had any effect on the existing unit tests one way or the other!

**Comment by albertored** [comment]
Yes, it's strange.. I'll take a look later

**Comment by albertored** [comment]
Now it should work, in tests history is always empty so I was trying to remove an element at index -1 and this raise a `ValueError` that is caught by the except statement in `onecmd_plus_hooks` method.

File: cmd2.py [hunk 1]
```diff
@@ -2046,6 +2046,9 @@ def select(self, opts, prompt='Your choice? '):
             self.poutput('  %2d. %s\n' % (idx + 1, text))
         while True:
             response = sm.input(prompt)
+            hlen = readline.get_current_history_length()
+            if hlen >= 1 and response != '':
+                readline.remove_history_item(hlen - 1)
```
Comment (tleonhardt) [inline cmd2.py]: Looks good.  Thanks for this fix.  Not polluting the readline history when using ``select()`` is a nice fix.

**Pull request merged**